test/functional/feature_rare_output_scripts.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Check how rare output script types are treated regarding
+   standardness and consensus-validity.
+"""
+import random
+
+from test_framework.blocktools import (
+    NORMAL_GBT_REQUEST_PARAMS,
+    create_block,
+)
+from test_framework.key import ECPubKey
+from test_framework.messages import (
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.script import sign_input_legacy
+from test_framework.script_util import (
+    key_to_p2pk_script,
+    keys_to_multisig_script,
+    output_key_to_p2tr_script,
+    program_to_witness_script,
+)
+from test_framework.crypto.secp256k1 import GE
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+from test_framework.wallet import MiniWallet
+from test_framework.wallet_util import generate_keypair
+
+
+class RareOutputScripts(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def generate_uncompressed_keypair(self, *, is_even):
+        while True:
+            privkey, pubkey_bytes = generate_keypair(compressed=False)
+            pubkey = ECPubKey()
+            pubkey.set(pubkey_bytes)
+            if pubkey.p.y.is_even() == is_even:
+                return privkey, pubkey
+
+    def create_p2pk_spend(self, outpoint, scriptpubkey, privkey):
+        tx = CTransaction()
+        tx.vin = [CTxIn(COutPoint(int(outpoint[0], 16), outpoint[1]))]
+        tx.vout = [CTxOut(40000, self.wallet.get_scriptPubKey())]
+        sign_input_legacy(tx, 0, scriptpubkey, privkey)
+        return tx
+
+    def create_empty_witness_tx(self, outpoint):
+        tx = CTransaction()
+        tx.vin = [CTxIn(COutPoint(int(outpoint[0], 16), outpoint[1]))]
+        tx.vout = [CTxOut(40000, self.wallet.get_scriptPubKey())]
+        tx.wit.vtxinwit.append(CTxInWitness())
+        return tx
+
+    def test_hybrid_pubkeys(self):
+        self.log.info("Check that creating hybrid pubkey P2PK outputs is standard")
+        privkey_even, pubkey_even = self.generate_uncompressed_keypair(is_even=True)
+        assert pubkey_even.p.y.is_even()
+        pubkey_even_script = key_to_p2pk_script(b'\x06' + pubkey_even.get_bytes()[1:])
```

Comment:
Thinking out loud:
Might be nice to enhance ECPubKey to handle hybrid keys (prevents having to do byte manipulation here).  Seems outside the scope of this PR though.

Reply:
not sure it's worth the effort for such a niche thing that already has unit test coverage